TACC/tup-ui
# task/TUP-196 -- client login component

## Overview:

Login component refactor, plus code improvements

## Related:

- [TUP-196](https://jira.tacc.utexas.edu/browse/TUP-196)

## Changes:

- Added `@tacc/tup-ui` to tsconfig paths and aliases for vite (tup-ui must use relative imports, see below)
- Add better test helpers to reduce boilerplate in tests
- Full page layout for application
- Formik, layout and styles for Login screen (see notes)

## Testing:

1. npm ci
2. nx serve tup-ui
3. Try logging in
4. Try bad username and password
5. Try setting an invalid [baseURL in index.html](https://github.com/TACC/tup-ui/blob/5f0cb079939f2f99ce790794068babb185f51a20/apps/tup-ui/index.html#L10) from tup-ui and then try logging in to see a server side error message

## UI:

Adobe XD: https://xd.adobe.com/view/860ae681-198f-4eee-802e-f1b97334cdce-d359/specs/

![image](https://user-images.githubusercontent.com/17181582/181149815-b7d7f42f-8732-4670-81b9-2d86fcfd6f21.png)


## Notes:

NX Notes @jarosenb (can be addressed in a subsequent PR to https://github.com/TACC/tup-ui/pull/40)

- Although I was able to get tsconfig and vite.config.ts to recognize `@tacc/tup-ui` as a module, NX specifically disallows this in the  @nrwl/nx/enforce-module-boundaries` eslint plugin. This is because nx requires these module boundaries as well as some additional project graph settings we are not using to do its "affected" project detection. I'm open to some minor restructuring of the directory tree to reduce the amount of `../../../hooks` everywhere.

UI Notes @wesleyboar (can be addressed in a subsequent PR to https://github.com/TACC/tup-ui/pull/40)

- Things that would be links in the mockup currently are not, since there are no valid destinations
- Got a quick mockup from Matt S on how error messages should look, approximately
- The exact font is wrong and I require assistance on getting it to work from `core-styles`
- The Loading state on the Log In button exceeds the width of the button itself. I believe this is a deficiency in the Button component's preset fixed widths, as I have glanced at and seen other examples (such as APCD form submission) where the Button text gets truncated
